test/functional/mochitest/browser_as_private_browsing.js

```diff
@@ -0,0 +1,30 @@
+"use strict";
+
+let Cu = Components.utils;
+Cu.import("resource://gre/modules/Task.jsm");
+Cu.import("resource://gre/modules/Services.jsm");
+
+/**
+ * Tests that loading activity streams in private browsing mode doesn't actually
+ * load the activity streams URL and instead loads about:privatebrowsing
+ */
+add_task(function*doesnt_load_in_private_windows() {
+  const asURL = "resource://activity-streams/data/content/activity-streams.html#/";
+  const privateBrowsingURL = "about:privatebrowsing";
+
+  // create a private browsing window
+  let privateWindow = yield BrowserTestUtils.openNewBrowserWindow({private: true});
+  let tab = privateWindow.gBrowser.selectedBrowser;
+
+  // try to load activity stream's url into the private window
+  tab.loadURI(asURL);
```

Comment:
https://wiki.mozilla.org/Electrolysis/e10s_test_tips seems to suggest that the right thing to do is use BrowserTestUtils.loadURI on the XUL browser element so that we don't run in to synchronicity problems which could cause intermittent failures.